pkgs/applications/audio/amarok/default.nix

```
@@ -3,21 +3,21 @@
 , qca-qt5, qjson, qtquickcontrols2, qtscript, qtwebengine
 , karchive, kcmutils, kconfig, kdnssd, kguiaddons, kinit, kirigami2, knewstuff, knotifyconfig, ktexteditor, kwindowsystem
 , fftw, phonon, plasma-framework, threadweaver
-, curl, ffmpeg, gdk-pixbuf, libaio, libmtp, loudmouth, lzo, lz4, mysql57, pcre, snappy, taglib, taglib_extras
+, curl, ffmpeg, gdk-pixbuf, libaio, liblastfm, libmtp, loudmouth, lzo, lz4, mysql57, pcre, snappy, taglib, taglib_extras
 }:
 
 mkDerivation rec {
   pname = "amarok";
-  version = "2.9.0-20190824";
+  version = "2.9.0-20200517";
```

Unless this is the real version used by the upstream, nixpkgs convention is this:
```suggestion
  version = "unstable-2020-05-17";
```

---

Thanks for the heads up. I found the following verbiage [in the nixpkgs manual](https://nixos.org/nixpkgs/manual/#sec-package-naming):

>If a package is not a release but a commit from a repository, then the version part of the name *must* be the date of that (fetched) commit. The date must be in `"YYYY-MM-DD"` format. Also append `"unstable"` to the name - e.g., `"pkgname-unstable-2014-09-23"`.

I think this means I should have:

```nix
  pname = "amarok-unstable";
  version = "2020-05-17";
```

No?

Grepping through nixpkgs, I see there are a bunch of matches for `name = .*unstable"`, but also a bunch of matches for `version = "unstable-`. It would probably be good if this convention were more standardized.

---

The documentation sadly says nothing about `pname` or `version`, it just says the resulting `name` should be in format `foo-unstable-YYYY-MM-DD`. Unfortunately, there are more ways how to achieve that.

See my comment here about options being discussed: https://github.com/NixOS/nixpkgs/pull/88023#issuecomment-629844186

I am fine with both options.